src/libs/ReportUtils.js

```diff
@@ -1111,6 +1131,22 @@ function isWaitingForIOUActionFromCurrentUser(report) {
         return false;
     }
 
+    const policy = getPolicy(report.policyID);
+    if (policy.type === CONST.POLICY.TYPE.CORPORATE) {
+        // If the report is already settled, there's no action required from any user.
+        if (isSettled(report.reportID)) {
+            return false;
+        }
+
+        // Report is pending approval and the current user is the manager
+        if (isReportManager(report) && !isReportApproved(report)) {
+            return true;
+        }
+
+        // Current user is an admin and the report has been approved but not settled yet
+        return policy.role === CONST.POLICY.ROLE.ADMIN && isReportApproved(report);
```

Comment:
I think this may be missing the case where the employee is missing a bank account and the report gets the RNVP `isWaitingOnBankAccount `. For CORPORATE policies, I think we should also return `true` if the report has `isWaitingOnBankAccount` and the current user is the submitter, right?

Reply:
And we should return false if `report.isWaitingOnBankAccount` is truthy and the current user is NOT the submitter because the we are expecting an action on the submitter. What actions could the admin or manager take?